### ISSUE TITLE
Loss obtained from callback different from model.evaluate()

### ISSUE THREAD

**Issue by brunoparis01** [keras-team/keras#8940]
Hello.

I am trying to build a custom loss function and making sure the value it gives is correct.
In order to do that, in the callback     **def on_batch_end(self, batch, logs={})** I do three checks:

1. I call self.model.evaluate() witch the batch_data
2. I calculate manually the loss from the batch_data and the predicted data
3. I obtain the loss from the variable logs['loss']

The value obtained in number # 1 and # 2 is the same.... however the loss obtained from # 3 is totally different from other ones. Why is that? is there some weight involved in the # 3 process? Can anybody help me on that?

Thanks very much.

**Comment by guifereis** [comment]
To extend #3, could you check if #3 is the same as the History object that fit() returns? Because I also see that that value is different from what I get with #1. My current theory is that it is because the evaluation is conducted after each epoch of training whereas the logs/history is calculated per minibatch *during* the epoch. or otherwise, perhaps due to image augmentation/generators...

**Comment by endolith** [comment]
@guifereis Did you find out why these are different?

**Comment by guifereis** [comment]
@endolith it has been a while, I believe so but can't recall what exactly it was. I think I was on the right track, look at what data each of them is evaluating over. One could be transformed/augmented but not the other..... or it could have been something subtler like the order of the calculations.
